# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/PDMFactory.cpp
# Commit: 1e465667a9d0
# Full Hash: 1e465667a9d0e0a48095b78739f9d05640a1f563
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P17. Only initialize PDM if enabled. r=mattwoodrow
#   
#   Attempting to initialize a PDM can cause the process to be killed due to sandbox violations.
#   For now, only FFVPX decoders is working and enabled in the RDD process.
#   
# ==============================================================================

diff -r 0e4fc20bf2d3 -r 1e465667a9d0 dom/media/platforms/PDMFactory.cpp
--- a/dom/media/platforms/PDMFactory.cpp	Tue Oct 20 23:28:32 2020 +0000
+++ b/dom/media/platforms/PDMFactory.cpp	Tue Oct 20 23:28:40 2020 +0000
@@ -68,10 +68,14 @@
     OmxDecoderModule::Init();
 #endif
 #ifdef MOZ_FFVPX
-    FFVPXRuntimeLinker::Init();
+    if (!XRE_IsRDDProcess() || StaticPrefs::media_rdd_ffvpx_enabled()) {
+      FFVPXRuntimeLinker::Init();
+    }
 #endif
 #ifdef MOZ_FFMPEG
-    FFmpegRuntimeLinker::Init();
+    if (!XRE_IsRDDProcess() || StaticPrefs::media_rdd_ffmpeg_enabled()) {
+      FFmpegRuntimeLinker::Init();
+    }
 #endif
   }
 };
